cpp/src/arrow/flight/types.cc

```diff
@@ -638,6 +726,11 @@ arrow::Result<std::unique_ptr<Result>> SimpleResultStream::Next() {
   return std::make_unique<Result>(std::move(results_[position_++]));
 }
 
+std::string BasicAuth::ToString() const {
+  return arrow::util::StringBuilder("BasicAuth<username = '", username, "', password = '",
+                                    password, "'>");
```

Comment:
It's dangerous to include the password here, as it could leak unwillingly through logging or other means.